src/Microsoft.OData.Client/DataServiceContext.cs

```diff
@@ -2567,6 +2596,25 @@ public virtual void UpdateObject(object entity)
             this.UpdateObjectInternal(entity, false /*failIfNotUnchanged*/);
         }
 
+        /// <summary>
+        /// Convert array object to List of strings.
+        /// </summary>
+        /// <param name="dependsOnIds">Object array of DependsOnIds.</param>
+        /// <returns>List of strings.</returns>
+        private static List<string> ConvertDependOnIdsToString(object[] dependsOnIds)

```

Comment:
since this list is not expanded could we switch this out to something that won't allocate any list. Since we are only doing conversion I believe yield return id.toString() could give us better performance memory wise as it had no allocations. 
```suggestion
        private static IEnumerable<string> ConvertDependOnIdsToString(object[] dependsOnIds)

```

Reply:
If we do make this into an IEnumerable `this.UpdateObjectInternal(entity, false /*failIfNotUnchanged*/, dependsOn.Count > 0 ? dependsOn : null);` this check which is done after the conversion could be done using the dependsOnIds.

Reply:
Once I changed the design, this method was removed